config/kubernetes/base/operator.yaml

```diff
@@ -63,3 +68,26 @@ spec:
             configMapKeyRef:
               name: tekton-config-defaults
               key: DEFAULT_TARGET_NAMESPACE
+      - name: tekton-operator-cluster-operations
+        image: ko://github.com/tektoncd/operator/cmd/kubernetes/operator
+        args:
+        - "-controllers"
+        - "tektoninstallerset"
+        - "-unique-process-name"
```

Comment:
what is `unique-process-name` :thinking:

Reply:
the value of this flag will be used as the "name" arguement in the call to knative/pkg sharedMain. 

knative/pkg uses this name ([component string](https://github.com/knative/pkg/blob/4d62e1dbb943ce606deb3820c62dfbcdaaa5118b/injection/sharedmain/main.go#L154)) to setup [logging](https://github.com/knative/pkg/blob/4d62e1dbb943ce606deb3820c62dfbcdaaa5118b/injection/sharedmain/main.go#L205) and [leader election](https://github.com/knative/pkg/blob/4d62e1dbb943ce606deb3820c62dfbcdaaa5118b/injection/sharedmain/main.go#L227).

In the current `main` branch, it is hard coded to "tekton-operator" [here](https://github.com/tektoncd/operator/blob/6c46be0331fcb6409107a8277e47374f4f2814a7/cmd/kubernetes/operator/main.go#L32).

Harcoding is fine as long as there is only 1 container. When there are multiple instances of operator controller running in different containers we need to make it unique. Otherwise, knative/pkg will setup common logging and leader election for those containers as a group.

i shall add a comment above the flag declaration. [here](https://github.com/nikhil-thomas/tektoncd-operator/blob/e5c458b871f370b4eacff5307ef1817712ef7b4b/pkg/reconciler/platform/config.go#L39)🧑‍💻